plugins/numbers/numbers.cpp

```diff
@@ -92,10 +92,21 @@ QVariant Numbers::data(const QModelIndex & index)
 
 void Numbers::setEditorData(QWidget * editor, const QModelIndex & /*index*/)
 {
-	if (value().type() == QVariant::Double)
-		dynamic_cast<QDoubleSpinBox *>(editor)->setValue(value().toDouble());
-	else
+	switch (value().type())
+	{
+	case QVariant::Double:
+	{
+		QString currentValue;
+		currentValue.setNum(value().toDouble(), 'g', 30);
+		dynamic_cast<QLineEdit *>(editor)->setText(currentValue);
+		break;
+	}
+	case QVariant::String:
+		dynamic_cast<QLineEdit *>(editor)->setText(value().toString());
+		break;
+	default:
 		dynamic_cast<QSpinBox *>(editor)->setValue(value().toDouble());
```

Comment:
Why is there still a spin box here?

Reply:
It is present only for integral values. There are integral and floating-point numbers supported in the numbers plugin for the property editor.